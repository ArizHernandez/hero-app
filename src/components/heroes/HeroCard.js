import React from 'react'
import { Link } from 'react-router-dom'
import { heroImages } from '../../helpers/heroImages'

export const HeroCard = ({ 
  alter_ego,
  characters,
  first_appearance,
  id,
  superhero
}) => {

  return (
    <div className="card text-center">
      <img 
        src={heroImages(id)} 
        className="card-img-top" 
        alt="..." 
      />

      <div className="card-body">
        <h5 className="card-title">{ superhero }</h5>
        <p className="card-text hero-name">
          {
            ( alter_ego !== characters )
            ? characters
            : alter_ego 
          }
        </p> 
        <p className="card-text text-end mb-2">
          <small className="text-muted">{ first_appearance }</small>
        </p>

        <Link
          to={`./hero/${ id }`}
          className="btn btn-dark btn-block"
        >
          Mas...
        </Link>
      </div>
    </div>
  )
}
