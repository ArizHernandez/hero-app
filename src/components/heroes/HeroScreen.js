import React, { useMemo } from 'react'
import { Redirect, useParams } from 'react-router'

import { getHerosById } from '../../selectors/getHerosById';
import { heroImages } from '../../helpers/heroImages';

export const HeroScreen = ({ history }) => {

  const {heroId} = useParams();
  const hero = useMemo(() => getHerosById({id: heroId}), [heroId]);

  if(!hero || hero.length === 0) {return <Redirect to="/"/>}

  const handleReturn = () => {
    if(history.length <= 2){
      history.push('/')
    } else {
      history.goBack();
    }

  } 

  const [{ superhero, publisher, alter_ego, first_appearance, characters }]  = hero;

  return (
    <div className="container mt-3">
      <button 
        className="btn btn-danger mb-4"
        onClick={ handleReturn }
      >
        Return
      </button>

      <div className="row">
        <div className="shadow-lg p-0 col-12 col-md-4 col-lg-3 animate__animated animate__backInLeft">
          <img 
            className="img-fluid rounded" 
            src={heroImages(heroId)} 
            alt={heroId}
          />
        </div>
        <div className="col text-center">
          <h3 className="display-4 mb-2">{ superhero } <small className="text-muted fs-4"> { publisher }&copy; </small> </h3>
          <p className="mb-4"> { alter_ego } - { first_appearance } </p>
          
          <h3 className="display-6 "> Characters </h3>
          <p> { characters } </p>
        </div>
      </div>
    </div>
  )
}
