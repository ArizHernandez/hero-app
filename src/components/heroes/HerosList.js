import React, { useMemo } from 'react'
import { getHerosByPublisher } from '../../selectors/getHerosByPublisher'
import { HeroCard } from './HeroCard'

export const HerosList = ({ publisher }) => {

  const heroes = useMemo(() => getHerosByPublisher({ publisher }), [ publisher ]);

  return (
    <div className="card-group">
      <div className="row row-cols-1 row-cols-md-3 row-cols-lg-5 g-4">
        { 
          heroes.map( hero => {
            return (
            <div 
              className=" animate__animated animate__fadeIn"
              key={ hero.id }
            >
              <HeroCard { ...hero }/>
            </div>
            )}
          ) 
        }
      </div>
    </div>
  )
}
