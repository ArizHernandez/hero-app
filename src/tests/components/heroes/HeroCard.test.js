import { shallow } from "enzyme"
import { HeroCard } from "../../../components/heroes/HeroCard"

describe('Pruebas en <HeroCard />', () => {
  
  const hero = {
    'id': 'dc-batman',
    'superhero':'Batman', 
    'publisher':'DC Comics', 
    'alter_ego':'Bruce Wayne',
    'first_appearance':'Detective Comics #27',
    'characters':'Bruce Wayne'
  }

  const wrapper = shallow(<HeroCard {...hero} />)

  test('Debe cargar correctamente', () => {
    expect(wrapper).toMatchSnapshot();
  });
  
  test('Debe de tener el nombre de Batman', () => {
    expect(wrapper.find('.card-title').text().trim()).toBe(hero.superhero);
  });
  
  test('Debe de tener solo alter_ego', () => {
    expect(wrapper.find('.hero-name').text().trim()).toBe(hero.alter_ego);
  });
  
  test('Debe de tener solo characters', () => {
    const hero = {
      'id': 'dc-flash',
      'superhero':'Flash', 
      'publisher':'DC Comics', 
      'alter_ego':'Jay Garrick',
      'first_appearance':'Flash Comics #1',
      'characters':'Jay Garrick, Barry Allen, Wally West, Bart Allen'
    }
    const wrapper = shallow(<HeroCard {...hero} />);

    expect(wrapper.find('.hero-name').text().trim()).toBe(hero.characters);
  }); 
})
