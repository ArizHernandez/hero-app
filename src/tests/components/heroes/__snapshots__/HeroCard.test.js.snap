// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <HeroCard /> Debe cargar correctamente 1`] = `
<div
  className="card text-center"
>
  <img
    alt="..."
    className="card-img-top"
    src=""
  />
  <div
    className="card-body"
  >
    <h5
      className="card-title"
    >
      Batman
    </h5>
    <p
      className="card-text hero-name"
    >
      Bruce Wayne
    </p>
    <p
      className="card-text text-end mb-2"
    >
      <small
        className="text-muted"
      >
        Detective Comics #27
      </small>
    </p>
    <Link
      className="btn btn-dark btn-block"
      to="./hero/dc-batman"
    >
      Mas...
    </Link>
  </div>
</div>
`;
